src/sql/src/statements/show.rs

```diff
@@ -74,18 +149,72 @@ pub struct ShowTables {
     pub full: bool,
 }
 
+impl ShowTables {
+    pub fn kind(&self) -> &ShowKind {
+        &self.kind
+    }
+
+    #[inline]
+    fn format_database(&self) -> String {
+        match &self.database {
+            None => String::default(),
+            Some(database) => {
+                format!("IN {}", database)
+            }
+        }
+    }
+}
+
+impl Display for ShowTables {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("SHOW")?;
+        if self.full {
+            f.write_str(" FULL")?;
+        }
+        let database = self.format_database();
+        let kind = self.kind();
+        write!(f, r#" TABLES {database} {kind}"#)
+    }
+}
+
 /// SQL structure for `SHOW CREATE TABLE`.
 #[derive(Debug, Clone, PartialEq, Eq, Visit, VisitMut)]
 pub struct ShowCreateTable {
     pub table_name: ObjectName,
 }
 
+impl ShowCreateTable {
+    fn table_name(&self) -> &ObjectName {
+        &self.table_name
+    }
+}
+
+impl Display for ShowCreateTable {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let table_name = self.table_name();
+        write!(f, r#"SHOW CREATE TABLE {table_name}"#)
+    }
+}
+
 /// SQL structure for `SHOW VARIABLES xxx`.
 #[derive(Debug, Clone, PartialEq, Eq, Visit, VisitMut)]
 pub struct ShowVariables {
     pub variable: ObjectName,
 }
 
+impl ShowVariables {
+    fn variable(&self) -> &ObjectName {
+        &self.variable
+    }
+}
+
+impl Display for ShowVariables {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let variable = self.variable();
+        write!(f, r#"SHOW VARIABLES {variable}"#)
+    }
+}
```

Comment:
Let's try to inline everywhere unless it's too complex to extract to a method. Such an indirection doesn't help much but introduce extra understanding burden.